src/qgis_geonode/gui/connection_dialog.py

```diff
@@ -1,84 +1,91 @@
+from qgis.core import Qgis
+from qgis.gui import QgsMessageBar
+from qgis.PyQt.QtWidgets import (
+    QDialog,
+    QDialogButtonBox,
+    QSizePolicy,
+)
+from qgis.PyQt.QtGui import QRegExpValidator
+from qgis.PyQt.QtCore import QRegExp
+from qgis.PyQt.uic import loadUiType
 
-from qgis_geonode.qgisgeonode.utils import tr
+from ..apiclient.api_client import GeonodeClient
+from ..qgisgeonode.conf import ConnectionSettings, connections_manager
 
 DialogUi, _ = loadUiType(
-    os.path.join(os.path.dirname(__file__), "../ui/qgis_geonode_connection.ui")
+    os.path.join(os.path.dirname(__file__), "../ui/connection_dialog.ui")
 )
 
 
 class ConnectionDialog(QDialog, DialogUi):
-    def __init__(self):
-        super(ConnectionDialog, self).__init__()
-        self.setupUi(self)
-        self.settings = QgsSettings()
-        self.connection_name = None
-        self.base_group = "/{}".format(SETTINGS_GROUP_NAME)
+    connection_id: uuid.UUID
+    geonode_client: GeonodeClient = None
 
+    def __init__(self, connection_settings: typing.Optional[ConnectionSettings] = None):
+        super().__init__()
+        self.setupUi(self)
+        self.bar = QgsMessageBar()
+        self.bar.setSizePolicy(QSizePolicy.Minimum, QSizePolicy.Fixed)
+        self.layout().insertWidget(0, self.bar)
+        if connection_settings is not None:
+            self.connection_id = connection_settings.id
+            self.load_connection_settings(connection_settings)
+        else:
+            self.connection_id = uuid.uuid4()
         self.buttonBox.button(QDialogButtonBox.Ok).setEnabled(False)
-
-        self.name.textChanged.connect(self.update_ok_button)
-        self.url.textChanged.connect(self.update_ok_button)
-
-        self.name.setValidator(QRegExpValidator(QRegExp("[^\\/]+"), self.name))
+        ok_signals = [
+            self.name_le.textChanged,
+            self.url_le.textChanged,
+        ]
+        for signal in ok_signals:
+            signal.connect(self.update_ok_buttons)
+        self.test_connection_btn.clicked.connect(self.test_connection)
+        # disallow names that have a slash since that is not compatible with how we
+        # are storing plugin state in QgsSettings
+        self.name_le.setValidator(QRegExpValidator(QRegExp("[^\\/]+"), self.name_le))
+        self.update_ok_buttons()
+
+    def load_connection_settings(self, connection_settings: ConnectionSettings):
+        self.name_le.setText(connection_settings.name)
+        self.url_le.setText(connection_settings.base_url)
+        self.authcfg_acs.setConfigId(connection_settings.auth_config)
+
+    def get_connection_settings(self) -> ConnectionSettings:
+        return ConnectionSettings(
+            id=self.connection_id,
+            name=self.name_le.text().strip(),
+            base_url=self.url_le.text().strip(),
+            auth_config=self.authcfg_acs.configId(),
+        )
+
+    def test_connection(self):
```

Comment:
we should disable the "Test Connection" button and enable it after finishing testing connection.

Reply:
:+1: I can add this in